asset/file_reader/slicer.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+
+const { Slicer } = require('@terascope/job-components');
+const Queue = require('@terascope/queue');
+const { getOffsets } = require('@terascope/chunked-file-reader');
+const { TSError } = require('@terascope/utils');
+const Promise = require('bluebird');
+const path = require('path');
+const fse = require('fs-extra');
+
+class FileSlicer extends Slicer {
+    constructor(context, opConfig, executionConfig) {
+        super(context, opConfig, executionConfig);
+        this._queue = new Queue();
+        this._doneSlicing = false;
+    }
+
+    async initialize() {
+        this.checkProvidedPath();
+        this.getFilePaths(this.opConfig.path)
+            .then(() => {
+                this._doneSlicing = true;
+            })
+            .catch((err) => {
+                // This will catch an error if there is  a problem getting the initial directory
+                // contents
+                const error = new TSError(err, { reason: `There was a problem slicing files from ${this.opConfig.path}: ${err}` });
```

Comment:
When you give TSError an error as input it will automatically add its context, either the stack or the message.
```suggestion
                const error = new TSError(err, { reason: `There was a problem slicing files from ${this.opConfig.path}` });
```

Reply:
removed redundant error message